server/game/cards/03-WC/BookOfLeq.js

```diff
@@ -0,0 +1,32 @@
+const Card = require('../../Card.js');
+
+class BookOfLeQ extends Card {
+    setupCardAbilities(ability) {
+        this.action({
+            gameAction: [
+                ability.actions.reveal(context => ({
+                    location: 'deck',
+                    target: context.player.deck.length > 0 ? context.player.deck[0] : []
+                })),
+                ability.actions.conditional(({
+                    condition: context => context.player.deck.length > 0 && context.player.deck[0].printedHouse !== 'staralliance',
+                    trueGameAction: ability.actions.changeActiveHouse(context => ({ house: context.player.deck[0].printedHouse })),
+                    falseGameAction: ability.actions.untilNextTurn({
+                        targetController: 'current',
+                        effect: [
+                            ability.effects.skipStep('ready'),
+                            ability.effects.skipStep('draw')
+                        ]
+                    }),
+                    postHandler: context => {
+                        context.game.endPhaseRightNow = context.player.deck.length > 0 && context.player.deck[0].printedHouse === 'staralliance';
```

Comment:
I think you need to check the status of the flag already - if somehow it's already true, you don't want to change it